aptos/core/src/crypto/sig.rs

```diff
@@ -120,7 +120,7 @@ impl PublicKey {
     /// A `Vec<u8>` representing the public key.
     pub fn to_bytes(&self) -> Vec<u8> {
         let mut bytes = BytesMut::new();
-        let pub_key_bytes = self.compressed_pubkey.as_ref();
+        let pub_key_bytes: &[u8] = self.compressed_pubkey.as_ref();
```

Comment:
This fixes a compile error when building with the current version of the succinct toolchain (v1.79):
```
    |
123 |         let pub_key_bytes = self.compressed_pubkey.as_ref();
    |             ^^^^^^^^^^^^^
124 |
125 |         bytes.put_slice(&write_leb128(pub_key_bytes.len() as u64));
    |                                                     --- type must be known at this point
    |
help: consider giving `pub_key_bytes` an explicit type, where the type for type parameter `T` is specified
    |
123 |         let pub_key_bytes: &T = self.compressed_pubkey.as_ref();
    |                          ++++
```